src/main/java/com/minecolonies/coremod/entity/ai/util/RecipeStorage.java

```diff
@@ -31,30 +54,56 @@
     private final List<ItemStack> secondaryOutput;
 
     /**
+     * The intermediate required for the recipe (e.g furnace).
+     */
+    private final Block intermediate;
+
+    /**
      * Grid size required for the recipe.
      */
     private final int gridSize;
 
     /**
      * Create an instance of the recipe storage.
-     * @param input the list of input items (required for the recipe).
-     * @param gridSize the required grid size to make it.
-     * @param primaryOutput the primary output of the recipe.
+     *
+     * @param input           the list of input items (required for the recipe).
+     * @param gridSize        the required grid size to make it.
+     * @param primaryOutput   the primary output of the recipe.
+     * @param secondaryOutput the secondary output (like buckets or similar).
+     */
+    public RecipeStorage(final List<ItemStack> input, final int gridSize, final ItemStack primaryOutput, final ItemStack... secondaryOutput)
+    {
+        this.input = Collections.unmodifiableList(input);
+        this.primaryOutput = primaryOutput;
+        this.secondaryOutput = Collections.unmodifiableList(Arrays.asList(secondaryOutput));
+        this.gridSize = gridSize;
+        this.intermediate = null;
+    }
+
+    /**
+     * Create an instance of the recipe storage.
+     *
+     * @param input           the list of input items (required for the recipe).
+     * @param gridSize        the required grid size to make it.
+     * @param primaryOutput   the primary output of the recipe.
+     * @param intermediate    the intermediate to use (e.g furnace).
      * @param secondaryOutput the secondary output (like buckets or similar).
      */
-    public RecipeStorage(final List<ItemStack> input, final int gridSize, final ItemStack primaryOutput, final ItemStack...secondaryOutput)
+    public RecipeStorage(final List<ItemStack> input, final int gridSize, final ItemStack primaryOutput, final Block intermediate, final ItemStack... secondaryOutput)
     {
         this.input = Collections.unmodifiableList(input);
         this.primaryOutput = primaryOutput;
         this.secondaryOutput = Collections.unmodifiableList(Arrays.asList(secondaryOutput));
         this.gridSize = gridSize;
+        this.intermediate = intermediate;
     }
 
     /**
      * Get the list of input items.
      * Suppressing Sonar Rule Squid:S2384
      * The rule thinks we should return a copy of the list and not the list itself.
      * But in this case the rule does not apply because the list is an unmodifiable list already
+     *
      * @return the list.
      */
     @SuppressWarnings("squid:S2384")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S2384' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)